src/libs/ReportUtils.ts

```diff
@@ -1975,6 +1978,60 @@ function isAwaitingFirstLevelApproval(report: OnyxEntry<Report>): boolean {
     return isProcessingReport(report) && submitsToAccountID === report.managerID;
 }
 
+function isAwaitingFirstLevelApprovalNew(report: OnyxEntry<Report>, reportActions: ReportAction[], policyParam?: OnyxEntry<Policy>): boolean {
+    if (!report) {
+        return false;
+    }
+
+    if (!isProcessingReport(report)) {
+        return false;
+    }
+
+    if (isIOUReportUsingReport(report)) {
+        return true;
+    }
+
+    // This will be fixed as part of https://github.com/Expensify/Expensify/issues/507850
+    // eslint-disable-next-line @typescript-eslint/no-deprecated
+    const policy = policyParam ?? getPolicy(report?.policyID);
+
+    if (hasDynamicExternalWorkflow(policy)) {
+        return false;
+    }
+
+    if (policy?.approvalMode === CONST.POLICY.APPROVAL_MODE.BASIC) {
+        return true;
+    }
+
+    const usedReportAction = isInstantSubmitEnabled(policy)
+        ? reportActions?.find((action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED)
+        : reportActions
+              ?.filter((action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.SUBMITTED)
+              ?.sort((a, b) => {
+                  if (!a.created || !b.created) {
+                      return !a.created ? 1 : -1;
+                  }
+                  return a.created < b.created ? 1 : -1;
+              })
+              ?.at(0);
```

Comment:
NAB we could declare a variable with the action name and DRY the find logic, e.g.
```suggestion
    const actionNameToUse = isInstantSubmitEnabled(policy) ? CONST.REPORT.ACTIONS.TYPE.CREATED : CONST.REPORT.ACTIONS.TYPE.SUBMITTED;
    const usedReportAction = reportActions?.filter((action) => action.actionName === actionNameToUse)
              ?.sort((a, b) => {
                  if (!a.created || !b.created) {
                      return !a.created ? 1 : -1;
                  }
                  return a.created < b.created ? 1 : -1;
              })
              ?.at(0);
```